src/components/reusable/dropdownAddNewBtn/dropdownAddNewBtn.ts

```diff
@@ -0,0 +1,1166 @@
+import { LitElement, html } from 'lit';
+import {
+  customElement,
+  property,
+  state,
+  query,
+  queryAssignedElements,
+} from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import DropdownScss from './dropdownAddNewBtn.scss';
+import './dropdownAddNewBtnOption';
+import '../tag';
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+import downIcon from '@carbon/icons/es/chevron--down/24';
+import errorIcon from '@carbon/icons/es/warning--filled/24';
+import clearIcon from '@carbon/icons/es/close/24';
+// import clearIcon16 from '@carbon/icons/es/close/16';
+import addIcon24 from '@carbon/icons/es/add/16';
+import '../textInput';
+import '@kyndryl-design-system/shidoka-foundation/components/button/button';
+
+/**
+ * Dropdown, single select.
+ * @fires on-change - Captures the input event and emits the selected value and original event details.
+ * @fires on-search - Capture the search input event and emits the search text.
+ * @fires on-clear-all - Captures the the multi-select clear all button click event and emits the value.
+ * @slot unnamed - Slot for dropdown options.
+ * @slot label - Slot for input label.
+ */
+@customElement('kyn-dropdown-addnew-btn')
```

Comment:
Again here, using kyn-dropdown directly rather than duplicating it as a base would probably be a better approach. You're getting more in to Pattern territory now vs the component level.